theano/tensor/tests/test_opt.py

```diff
@@ -3135,6 +3135,156 @@ def test_local_fill_useless():
+         |Y [@K]
+
+        Inner graphs of the scan ops:
+
+        for{cpu,scan_fn} [@B] ''
+         >Sum{acc_dtype=float64} [@L] ''
+         > |X[t] [@M] -> [@I]
+        """
+        
+        mode = theano.compile.get_default_mode().excluding('fusion')
+        f = theano.function([X, Y], Z, mode=mode)
+        theano.printing.debugprint(f, print_type=True)
+        # here is the output for the debug print:
+        """
+        Elemwise{Add}[(0, 0)] [@A] <TensorType(float64, vector)> ''   7
+         |for{cpu,scan_fn} [@B] <TensorType(float64, vector)> ''   6
+         | |Shape_i{0} [@C] <TensorType(int64, scalar)> ''   0
+         | | |X [@D] <TensorType(float64, matrix)>
+         | |Subtensor{int64:int64:int8} [@E] <TensorType(float64, matrix)> ''   5
+         | | |X [@D] <TensorType(float64, matrix)>
+         | | |ScalarFromTensor [@F] <int64> ''   4
+         | | | |Elemwise{switch,no_inplace} [@G] <TensorType(int64, scalar)> ''   3
+         | | |   |Elemwise{le,no_inplace} [@H] <TensorType(int8, scalar)> ''   2
+         | | |   | |Shape_i{0} [@C] <TensorType(int64, scalar)> ''   0
+         | | |   | |TensorConstant{0} [@I] <TensorType(int8, scalar)>
+         | | |   |TensorConstant{0} [@I] <TensorType(int8, scalar)>
+         | | |   |TensorConstant{0} [@J] <TensorType(int64, scalar)>
+         | | |ScalarFromTensor [@K] <int64> ''   1
+         | | | |Shape_i{0} [@C] <TensorType(int64, scalar)> ''   0
+         | | |Constant{1} [@L] <int8>
+         | |Shape_i{0} [@C] <TensorType(int64, scalar)> ''   0
+         |Y [@M] <TensorType(float64, vector)>
+
+        Inner graphs of the scan ops:
+
+        for{cpu,scan_fn} [@B] <TensorType(float64, vector)> ''
+         >Sum{acc_dtype=float64} [@N] <TensorType(float64, scalar)> ''
+         > |X[t] [@O] <TensorType(float64, vector)> -> [@E]
+        """
+
+    def test_inequality_with_self(self):
+        x = T.scalar('x', dtype=config.floatX)
+        mode = theano.compile.get_default_mode().including('local_useless_elemwise_comparison')
+        
+        f = theano.function([x], T.lt(x, x), mode=mode)
+        assert_eqs_const(f.maker.fgraph.toposort(), 0)
+
+        f = theano.function([x], T.le(x, x), mode=mode)
+        assert_eqs_const(f.maker.fgraph.toposort(), 1)
+
+        f = theano.function([x], T.gt(x, x), mode=mode)
+        assert_eqs_const(f.maker.fgraph.toposort(), 0)
+
+        f = theano.function([x], T.ge(x, x), mode=mode)
+        assert_eqs_const(f.maker.fgraph.toposort(), 1)
+
+    def assert_identity(self, f):
+        topo = f.maker.fgraph.toposort()
+        assert topo[0].op == deep_copy_op
+        x_val = -128
+        assert f(x_val) == x_val
+        x_val = -1
+        assert f(x_val) == x_val
+        x_val = 0
+        assert f(x_val) == x_val
+        x_val = 1
+        assert f(x_val) == x_val
+        x_val = 127
+        assert f(x_val) == x_val
+        x_val = numpy.random.randint(255)-128
+        assert f(x_val) == x_val
```

Comment:
with that extra assert, just keep one assert with value. All the rest would be duplicate.